snorkel/learning/gen_learning.py

```diff
@@ -184,55 +188,153 @@ def __init__(self, class_prior=False, lf_prior=False, lf_propensity=False, lf_cl
+    # as well as maps defined within methods, require manual adjustments to 
+    # implement changes.
     #
-    # These names are also used by other related classes, such as GenerativeModelParameters
+    # These names are also used by other related classes, such as 
+    # GenerativeModelParameters
     optional_names = ('lf_prior', 'lf_propensity', 'lf_class_propensity')
-    dep_names = ('dep_similar', 'dep_fixing', 'dep_reinforcing', 'dep_exclusive')
-
-    def train(self, L, y=None, deps=(), init_acc = 1.0, init_deps=1.0, init_class_prior=-1.0, epochs=10, step_size=None, decay=0.99, reg_param=0.1, reg_type=2, verbose=False,
-              truncation=10, burn_in=5, timer=None):
+    dep_names = (
+        'dep_similar', 'dep_fixing', 'dep_reinforcing', 'dep_exclusive'
+    )
+
+    def train(self, L, deps=(), LF_priors=None, LF_prior_default=0.7, 
+        labels=None, label_prior=0.95, init_deps=1.0, 
+        init_class_prior=-1.0, epochs=100, step_size=None, decay=0.99, 
+        reg_param=0.1, reg_type=2, verbose=False, truncation=10, burn_in=5,
+        timer=None):
         """
-        Fits the parameters of the model to a data set. By default, learns a conditionally independent model.
-        Additional unary dependencies can be set to be included in the constructor. Additional pairwise and higher-order
-        dependencies can be included as an argument.
+        Fits the parameters of the model to a data set. By default, learns a 
+        conditionally independent model. Additional unary dependencies can be 
+        set to be included in the constructor. Additional pairwise and 
+        higher-order dependencies can be included as an argument.
 
-        Results are stored as a member named weights, instance of snorkel.learning.gen_learning.GenerativeModelWeights.
+        Results are stored as a member named weights, instance of 
+        snorkel.learning.gen_learning.GenerativeModelWeights.
 
         :param L: labeling function output matrix
-        :param y: optional ground truth labels
-        :param deps: collection of dependencies to include in the model, each element is a tuple of the form
-                     (LF 1 index, LF 2 index, dependency type), see snorkel.learning.constants
-        :param init_acc: initial weight for accuracy dependencies (in log scale)
-        :param init_deps: initial weight for additional dependencies, except class prior (in log scale)
-        :param init_class_prior: initial class prior (in log scale), note only used if class_prior=True in constructor
+        :param deps: collection of dependencies to include in the model, each 
+                     element is a tuple of the form 
+                     (LF 1 index, LF 2 index, dependency type),
+                     see snorkel.learning.constants
+        :param LF_priors: Priors on the LF accuracies
+        :param LF_prior_default: Default prior for the LFs
```

Comment:
I don't understand what a default prior is, what happens if I don't set it, and how to specify non-default priors (whatever those are).